### ISSUE TITLE
Gathered 0 statistics

### ISSUE THREAD

**Issue by davydkov** [varnish/newrelic_varnish_plugin#2]
Hi guys

Just installed this plugin, but it does not send any information.
This is an output from plugin.

What should I check?

```
# ruby -v
ruby 1.8.7 (2011-06-30 patchlevel 352) [x86_64-linux]
# varnishd -V
varnishd (varnish-3.0.3 revision 9e6a70f)
Copyright (c) 2006 Verdens Gang AS
Copyright (c) 2006-2011 Varnish Software AS
# ./newrelic_varnish_plugin
Faraday: you may want to install system_timer for reliable timeouts
Agent Varnish is at version 0.0.4
can't convert Array into String: ./newrelic_varnish_plugin:78:in `popen'
  ./newrelic_varnish_plugin:78:in `varnishstat'
  ./newrelic_varnish_plugin:138:in `poll_cycle'
  /usr/lib/ruby/gems/1.8/bundler/gems/newrelic_plugin-99e5dcafd8ff/lib/newrelic_plugin/agent.rb:143:in `run'
  /usr/lib/ruby/gems/1.8/bundler/gems/newrelic_plugin-99e5dcafd8ff/lib/newrelic_plugin/run.rb:95:in `loop_forever'
  /usr/lib/ruby/gems/1.8/bundler/gems/newrelic_plugin-99e5dcafd8ff/lib/newrelic_plugin/run.rb:93:in `each'
  /usr/lib/ruby/gems/1.8/bundler/gems/newrelic_plugin-99e5dcafd8ff/lib/newrelic_plugin/run.rb:93:in `loop_forever'
  /usr/lib/ruby/gems/1.8/bundler/gems/newrelic_plugin-99e5dcafd8ff/lib/newrelic_plugin/run.rb:17:in `setup_and_run'
  ./newrelic_varnish_plugin:186
Gathered 0 statistics
```

**Comment by Rud5G** [comment]
Me too.

**Comment by tfheen** [comment]
]] Denis Davydkov

> Just installed this plugin, but it does not send any information.
> This is an output from plugin.

It looks like your ruby is too old, we use the 1.9 format.  I'll take a
look at this later, there's a pull request that should hopefully fix
this.

## 

Tollef Fog Heen
Technical lead | Varnish Software AS
t: +47 21 98 92 64
We Make Websites Fly!

**Comment by pdunnigan** [comment]
same ... 
Ubuntu 12.04.2 LTS
ruby 1.8.7 (2011-06-30 patchlevel 352) [x86_64-linux]

```
can't convert Array into String: ./newrelic_varnish_plugin:77:in `popen'
  ./newrelic_varnish_plugin:77:in `varnishstat'
  ./newrelic_varnish_plugin:137:in `poll_cycle'
  /var/lib/gems/1.8/gems/newrelic_plugin-1.0.3/lib/newrelic_plugin/agent.rb:143:in `run'
  /var/lib/gems/1.8/gems/newrelic_plugin-1.0.3/lib/newrelic_plugin/run.rb:95:in `loop_forever'
  /var/lib/gems/1.8/gems/newrelic_plugin-1.0.3/lib/newrelic_plugin/run.rb:93:in `each'
  /var/lib/gems/1.8/gems/newrelic_plugin-1.0.3/lib/newrelic_plugin/run.rb:93:in `loop_forever'
  /var/lib/gems/1.8/gems/newrelic_plugin-1.0.3/lib/newrelic_plugin/run.rb:17:in `setup_and_run'
  ./newrelic_varnish_plugin:185
```

**Comment by jamrok** [comment]
I had the same issue on Ubuntu 12.04.2 with ruby 1.87.

There's a simple fix for this that was staring me in the face and I didn't realize it:

Change line 77 in the file [ newrelic_varnish_plugin ] to "convert Array into String" for the "cmd" array
OLD Line 77:
`IO::popen(cmd, mode="r") do |io|`
NEW Line 77:
`IO::popen(cmd.join(" "), mode="r") do |io|`

Once you execute it again, you get the following and "Varnish" pops up on the left in the NewRelic Dashboard:

```
Agent Varnish is at version 0.0.4
Metrics for varnish[default][com.varnish-software.newrelic-stat] for last 60 seconds:
  Sent 7 metrics to New Relic [https://platform-api.newrelic.com]:
    Component/Number of objects/N struct object[objects]: count: 1, total: 2.0, min: 2.0, max: 2.0, sum_of_squares: 4.0
    Component/Memory usage/SMS outstanding bytes[bytes]: count: 1, total: 0.0, min: 0.0, max: 0.0, sum_of_squares: 0.0
    Component/Memory usage/SMF.s0 Bytes available[bytes]: count: 1, total: 1073717248.0, min: 1073717248.0, max: 1073717248.0, sum_of_squares: 1.1

    Component/Memory usage/SMF.s0 Bytes outstanding[bytes]: count: 1, total: 24576.0, min: 24576.0, max: 24576.0, sum_of_squares: 603979776.0
    Component/Memory usage/SMA.Transient Bytes available[bytes]: count: 1, total: 0.0, min: 0.0, max: 0.0, sum_of_squares: 0.0
    Component/Memory usage/SMA.Transient Bytes outstanding[bytes]: count: 1, total: 0.0, min: 0.0, max: 0.0, sum_of_squares: 0.0
    Component/Thread status/N worker threads[threads]: count: 1, total: 2.0, min: 2.0, max: 2.0, sum_of_squares: 4.0
Gathered 7 statistics
```

Hope this helps. 

I'm still new to git so if this doesn't get fixed by time I get some more time to play with git, I'll fork it and make updates and submit a pull request.

**Comment by tfheen** [comment]
This has been fixed now, please try the new version.
